ProteoformSuiteInternal/ResultsSummaryGenerator.cs

```diff
@@ -373,23 +425,115 @@ public static DataTable results_dataframe(TusherAnalysis analysis)
+                    Sweet.lollipop.significance_by_log2FC ? e.quant.Log2FoldChangeValues.denominatorIntensitySum : get_tusher_values(e.quant, analysis).denominatorIntensitySum,
                     Sweet.lollipop.significance_by_log2FC ? e.quant.Log2FoldChangeValues.significant : get_tusher_values(e.quant, analysis).significant
                 );
             }
+
+             StringBuilder result_string = new StringBuilder();
+            string header = "";
+            foreach (DataColumn column in results.Columns)
+            {
+                header += column.ColumnName + "\t";
+            }
+            result_string.AppendLine(header);
+            foreach (DataRow row in results.Rows)
+            {
+                result_string.AppendLine(String.Join("\t", row.ItemArray));
+            }
             return results;
         }
 
-        public static DataTable biological_replicate_intensities(IGoAnalysis analysis, IEnumerable<ExperimentalProteoform> proteoforms, List<InputFile> input_files, Dictionary<string, List<string>> conditionsBioReps, bool include_imputation)
+
+        public static DataTable topdown_results_dataframe()
+        {
+            DataTable results = new DataTable();
+            results.Columns.Add("PFR", typeof(string));
+            results.Columns.Add("Theoretiecal Accession", typeof(string));
+            results.Columns.Add("Top-Down Full Accession", typeof(string));
+            results.Columns.Add("Top-Down  Accession", typeof(string));
+            results.Columns.Add("Theoretical Description", typeof(string));
+            results.Columns.Add("Theoretical Begin and End", typeof(string));
+            results.Columns.Add("Top-Down Begin and End", typeof(string));
+            results.Columns.Add("Theoretical PTM Type", typeof(string));
+            results.Columns.Add("Top-Down PTM Type", typeof(string));
+            results.Columns.Add("Top-Down PTM Type Unlocalized", typeof(string));
+            results.Columns.Add("Proteoform Suite Mass Error", typeof(string));
+            results.Columns.Add("Top-Down Mass Error", typeof(double));
+            results.Columns.Add("Proteoform Mass");
+            results.Columns.Add("Retention Time", typeof(double));
+            results.Columns.Add("Best Scoring Hit", typeof(string));
+            results.Columns.Add("Theoretical SGD ID", typeof(string));
+            results.Columns.Add("Theoretical Gene Name", typeof(string));
+            results.Columns.Add("Top-Down Gene Name", typeof(string));
+            results.Columns.Add("Family ID", typeof(string));
+            results.Columns.Add("Correct ID", typeof(bool));
+            results.Columns.Add("Accepted", typeof(bool));
+
+            foreach (TopDownProteoform td in Sweet.lollipop.topdown_proteoforms)
+            {
+                results.Rows.Add(
+                    td.pfr,
```

Comment:
Yeah, if PFR is an accession, it would be nice to have this property be "PfrAccession"

Reply:
Good idea